[add] add Diffblue Deeptest tests to increase line coverage

petersear opened the pull request:
This is a trial-run of the PR, against the Diffblue fork, to allow us to review the tests before submission of the actual PR. The tests included were identified by test-validator and are those remaining after rejecting any which failed to compile. Suggestions and reasons for tests to leave out are welcome, as are any opinions on what is good about some of these tests. Including everyone in the CX team so we can all see the discussion.

(The following text and the heading of this "PR" are from the commit message in the format discussed with Peter and Paul recently)

The classes covered by these tests are:

FilledListTest
ForkJoinSumCalculatorTest
GenericVarargsTest
HolderTest
MyObject0Test
MyObject1Test
MyObject2Test
MyObject3Test
OptionalInActionTest
OptionalMainTest
ParallelStreamsTest
PartitionPrimeNumbersTest
SimpleClassTest
TraderTest
WildcardsTest

In order to ensure these tests run successfully, a selection of dependencies have been added to the POM file (Powermock, Mockito & Diffblue's own Deeptest Utils)

peterschrammel commented on java-base/src/test/java/com/brianway/learning/java/base/generics/GenericWritingTest.java:
```diff
@@ -0,0 +1,40 @@
+    Method methodUnderTest = classUnderTest.getDeclaredMethod(
+        "writeExact", Reflector.forName("java.util.List"), Reflector.forName("java.lang.Object"));
+    methodUnderTest.setAccessible(true);
+    methodUnderTest.invoke(null, list, item);
```
What is this testing?

peterschrammel commented on java-base/src/test/java/com/brianway/learning/java/base/generics/SuperTypeWildcardsTest.java:
```diff
@@ -0,0 +1,42 @@
+    Method methodUnderTest =
+        classUnderTest.getDeclaredMethod("writeTo", Reflector.forName("java.util.List"));
+    methodUnderTest.setAccessible(true);
+    methodUnderTest.invoke(null, apples);
```
What is this testing?

peterschrammel commented on java-jvm/src/test/java/com/brianway/learning/java/jvm/memory/JavaVMStackSOFTest.java:
```diff
@@ -0,0 +1,24 @@
+  public void constructorOutputVoid() {
+
+    // Act, creating object to test constructor
+    JavaVMStackSOF objectUnderTest = new JavaVMStackSOF();
```
This is not testing much.

peterschrammel commented on java-multithread/src/test/java/com/brianway/learning/java/concurrent/LiftOffTest.java:
```diff
@@ -0,0 +1,24 @@
+  public void constructorOutputVoid() {
+
+    // Act, creating object to test constructor
+    LiftOff objectUnderTest = new LiftOff();
```
This is not testing much.

peterschrammel commented on java-multithread/src/test/java/com/brianway/learning/java/multithread/communication/example1/MyListTest.java:
```diff
@@ -0,0 +1,36 @@
+    Reflector.setField(objectUnderTest, "list", arrayList);
+
+    // Act
+    objectUnderTest.add();
```
What is this testing?

peterschrammel commented on java-multithread/src/test/java/com/brianway/learning/java/multithread/communication/example4/Run4_waitHasParam2Test.java:
```diff
@@ -0,0 +1,37 @@
+    Object retval = (Object)methodUnderTest.invoke(null);
+
+    // Assert result
+    Assert.assertNotNull(retval);
```
That's a bridge method.

peterschrammel commented on java-multithread/src/test/java/com/brianway/learning/java/multithread/communication/example4/Run4_waitHasParamTest.java:
```diff
@@ -0,0 +1,37 @@
+    Object retval = (Object)methodUnderTest.invoke(null);
+
+    // Assert result
+    Assert.assertNotNull(retval);
```
bridge method

peterschrammel commented on java-multithread/src/test/java/com/brianway/learning/java/multithread/communication/example5/MyRunTest.java:
```diff
@@ -0,0 +1,47 @@
+    boolean retval = (boolean)methodUnderTest.invoke(null, x0);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
```
bridge method

peterschrammel commented on java-multithread/src/test/java/com/brianway/learning/java/multithread/synchronize/example2/HasSelfPrivateNumTest.java:
```diff
@@ -0,0 +1,30 @@
+    String username = "`aaaaaaaaaaaaaaaaa";
+
+    // Act
+    objectUnderTest.addI(username);
```
Assertion missing.

peterschrammel commented on java-multithread/src/test/java/com/brianway/learning/java/multithread/synchronize/example5/MainTest.java:
```diff
@@ -0,0 +1,24 @@
+  public void constructorOutputVoid() {
+
+    // Act, creating object to test constructor
+    Main objectUnderTest = new Main();
```
Not very useful in isolation

peterschrammel commented on pom.xml:
```diff
@@ -65,4 +65,23 @@
+            <version>1.6.0</version>
+        </dependency>
+    </dependencies>
 </project>
```
newline missing

petersear replied:
Fixed (there was no newline in the original file)

peterschrammel commented on pom.xml:
```diff
@@ -65,4 +65,23 @@
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-junit4</artifactId>
+            <version>1.6.6</version>
```
1.6.5 now

petersear replied:
Ah yes, I see we went back from 1.6.6.
Reloaded https://github.com/diffblue/platform/blob/develop/docs/user/topics/executing-diffblue-tests.md and it is correct now.

Commit:
[add] add Diffblue Deeptest tests to increase line coverage

    The classes covered by these tests are:

    FilledListTest
    ForkJoinSumCalculatorTest
    GenericVarargsTest
    HolderTest
    MyObject0Test
    MyObject1Test
    MyObject2Test
    MyObject3Test
    OptionalInActionTest
    OptionalMainTest
    ParallelStreamsTest
    PartitionPrimeNumbersTest
    SimpleClassTest
    TraderTest
    WildcardsTest

In order to ensure these tests run successfully, a selection of dependencies have been added to the POM file (Powermock, Mockito & Diffblue's own Deeptest Utils)

petersear commented:
I've updated the branch to fix the Powermock version and add a final newline in pom.xml.
Also removed the tests identified as "not interesting" in earlier comments and updated the commit message accordingly.

peterschrammel reviewed (approved):
Looks better now

peterschrammel commented:
What is the `[add]` prefix in the commit message?

petersear commented:
Thanks.
I looked at previous commits on this repository and copied the style including new items being prefixed with "[add]".

petersear commented:
PR was submitted to against the parent repository:
https://github.com/brianway/java-learning/pull/16
